src/renderers/WebGLRenderer.js

```diff
@@ -1483,12 +1762,80 @@ function WebGLRenderer( parameters ) {
 
 			} else {
 
-				_this.renderBufferDirect( camera, scene.fog, geometry, material, object, group );
+				var instanceCount = undefined;
+
+				if ( geometry && geometry.isInstancedBufferGeometry && geometry.maxInstancedCount > 0 ) {
+
+					instanceCount = geometry.maxInstancedCount;
+
+				}
+
+				_this.renderBufferDirect( camera, scene.fog, geometry, material, object, group, instanceCount );
+
+			}
+
+			if ( object.onAfterRender ) {
+
+				object.onAfterRender( _this, scene, camera, geometry, material, group );
 
 			}
 
-			object.onAfterRender( _this, scene, camera, geometry, material, group );
+		}
+
+	}
+
+	function renderInstancedGeometryMap ( instancedGeometryMap, scene, camera, overrideMaterial ) {
+
+		for ( var uuid in instancedGeometryMap ) {
+
+			if ( instancedGeometryMap.hasOwnProperty( uuid ) ) {
+
+				var instancedGeometry = instancedGeometryMap[ uuid ];
+				var geometry = instancedGeometry.geometry;
+
+				for (var groupIndex = 0, groupCount = instancedGeometry.groups.length; groupIndex < groupCount; groupIndex++ ) {
+
+					var instancingGroup = instancedGeometry.groups[ groupIndex ];
+
+					var objects = instancingGroup.objects;
+					var material = overrideMaterial ? overrideMaterial : instancingGroup.material;
+
+					if ( objects.length > 0 ) {
+
+						var group = geometry.groups[ groupIndex ] || null;
 
+						for ( var i = 0, l = objects.length; i < l; i++ ) {
+
+							var object = objects[ i ];
+
+							object.modelViewMatrix.multiplyMatrices( camera.matrixWorldInverse, object.matrixWorld );
+							object.normalMatrix.getNormalMatrix( object.modelViewMatrix );
+
+						}
+
+						if ( objects.length < _this.minInstancingBatchSize ) {
+
+							for ( var i = 0, l = objects.length; i < l; i++ ) {
+
+								_this.renderBufferDirect( camera, scene.fog, geometry, material, objects[ i ], group );
```

Comment:
OK I was close but not quite on the mark; the real root of the issue is further upstream, in `pushRenderItem`. The code there that builds up the instancedGeometryMap(s) assumes that there will always be a `groupIndex` passed in so it knows which of its own groups to push onto, but that's not the case -- `groupIndex` will be null if the material is not a multimaterial. This results in a `groups` collection that is the same length as the geometry's groups, but has all objects pushed into the first group; therefore all but the first geometry group are skipped because they have objects.length==0.

I'm testing a local fix and will issue a pull request against this branch shortly.